File: Algorithms/8. String to Integer (atoi)/php/solution.php
<?php
class Solution
{

    /**
     * @param String $s
     * @return Integer
     */
    function myAtoi($s)
    {
        $s = trim($s);
        $sign = 1;
        $result = 0;
        $i = 0;
        if (strlen($s) == 0) {
            return 0;
        }
        if ($s[0] == '-') {
            $sign = -1;
            $i = 1;
        } else if ($s[0] == '+') {
            $i = 1;
        }
        for (; $i < strlen($s); $i++) {
            if ($s[$i] < '0' || $s[$i] > '9') {
                break;
            }
            $result = $result * 10 + $s[$i] - '0';
            if ($result * $sign > 2147483647) {
                return 2147483647;
            }
            if ($result * $sign < -2147483648) {
                return -2147483648;
            }
        }
        return $result * $sign;
    }
}

File: Algorithms/8. String to Integer (atoi)/rust/solution.rs
impl Solution {
    pub fn my_atoi(s: String) -> i32 {
        let s = s.trim();
        let mut result = 0;
        let mut sign = 1;
        let mut i = 0;
        if s.len() == 0 {
            return 0;
        }
        if s.chars().nth(0) == Some('-') {
            sign = -1;
            i = 1;
        } else if s.chars().nth(0) == Some('+') {
            i = 1;
        }
        for c in s.chars().skip(i) {
            if c.is_ascii_digit() {
                result = result * 10 + (c as i32 - 48);
                if result > 2147483647 {
                    return 2147483647;
                }
            } else {
                break;
            }
        }
        return sign * result;
    }
}
